en/sql/jsp.rst

```diff
@@ -505,90 +704,429 @@ In the calling block, you must set the OUT argument with **Types.JAVA_OBJECT**,
+                System.err.println(e2.getMessage());
+            } finally {
+                if (stmt != null) stmt.close();
+                if (conn != null) conn.close();
             }
-            
-            v.add(aoid);
-            set[0] = (CUBRIDOID[]) v.toArray(new CUBRIDOID[]{});
-            
-        } catch(Exception e) {
-            e.printStackTrace();
-            System.err.println("SQLException:"+e.getMessage());
         }
     }
 
-Using OID in Java Stored Function/Procedure
--------------------------------------------
+When the Java stored function/procedure is executed, it should run only on JVM located in the database server, you can check where it is running by calling System.getProperty ("cubrid.server.version") from the Java program source. The result value is the database version if it is called from the database; otherwise, it is **NULL**.
 
-In case of using the OID type value for IN/OUT in CUBRID, use the value passed from the server.
+.. _jsp-load-java:
 
-.. code-block:: sql
+loadjava Utility
+================
 
-    CREATE PROCEDURE tOID(i inout object, q string) AS LANGUAGE JAVA
-    NAME 'OIDtest.tOID(cubrid.sql.CUBRIDOID[], java.lang.String)';
+To load a compiled Java or JAR (Java Archive) file into CUBRID, use the **loadjava** utility. If you load a Java \*.class or \*.jar file using the **loadjava** utility, the file is moved to the specified database path. ::
 
-.. code-block:: java
+    loadjava [option] database-name java-class-file
 
-    import java.sql.*;
-    import cubrid.sql.CUBRIDOID;
+*   *database-name*: The name of the database where the Java file to be loaded.
+*   *java-class-file*: The name of the Java class or jar file to be loaded.
+*   [*option*]
+
+    *   **-y**: Automatically overwrites a class file with the same name, if any. The default value is **no**. If you load the file without specifying the **-y** option, you will be prompted to ask if you want to overwrite the class file with the same name (if any).
+
+.. _jsp-caution:
+
+Caution
+=======
+
+* java.sql.DatabaseMetaData is not supported.
+* JDBC API related to BLOB/CLOB type is not supported.
+* Functions not related to query execution and used only in client-side JDBC are not supported. For details, refer to :ref:`jsp-appendix`\.
+* Multiple SQL statements are not supported when executing a query with one JDBC object.
+* ResultSet created by query execution is non-updatable, non-scrollable, and non-sensitive.
+* Java ignores the return value and type digits of IN/OUT parameters, matches only the type, and delivers it to the database as it is.
+* A stored procedure can call another stored procedure or call itself recursively. The maximum nesting depth is 16.
 
-    public static void tOID(CUBRIDOID[] oid, String query)
-    {
-        Connection conn = null;
-        Statement stmt = null;
-        String ret = "";
+Returning Value of Java Stored Function/Procedure and Precision Type on IN/OUT
```

Comment:
```suggestion
Length of Return Values
```

Reply:
I've revised it as follows
`Limitations on the precision of IN/OUT parameters and a return value`